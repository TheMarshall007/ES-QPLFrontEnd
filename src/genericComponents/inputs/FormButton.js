import { useFormFull } from "form-full";
import React from "react";
import { Button, Spinner } from "react-bootstrap";

function FormButtonComponent({
  children,
  disabled,
  loading,
  name,
  actionType,
  onClick: propsOnClick,
  ...props
}) {
  const { onClick, formLoading, formDisabled } = useFormFull.button({
    name,
    actionType,
    onClick: propsOnClick,
  });

  return (
    <Button
      {...props}
      disabled={disabled || formDisabled}
      type={null}
      onClick={loading || formLoading ? null : onClick}
    >
      {loading || formLoading ? (
        <Spinner animation="border" size="sm" />
      ) : (
        children
      )}
    </Button>
  );
}

export default FormButtonComponent;
