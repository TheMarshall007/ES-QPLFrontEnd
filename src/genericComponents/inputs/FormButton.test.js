import React from "react";
import { render, screen } from "@testing-library/react";
import FormButton from "./FormButton";
import userEvent from "@testing-library/user-event";
import { FormFull } from "form-full";

describe("genericComponents / inputs / <FormButton />", () => {
  it('should render button with children text "Testando Botão"', () => {
    render(<FormButton name="submit">Testando Botão</FormButton>);

    // número de "expect" que deram certo;
    expect.assertions(1);
    const button = screen.getByRole("button", { name: /testando botão/i });
    expect(button).toBeInTheDocument();
  });

  it("should call function on button click", () => {
    const fn = jest.fn();
    render(
      <FormButton name="submit" onClick={fn}>
        Testando Botão
      </FormButton>
    );

    const button = screen.getByRole("button", { name: /testando botão/i });
    userEvent.click(button);
    expect(fn).toHaveBeenCalledTimes(1);
  });

  it("should be disabled when disabled is true", () => {
    render(
      <FormButton name="submit" disabled>
        Testando Botão
      </FormButton>
    );
    const button = screen.getByRole("button", { name: /testando botão/i });
    expect(button).toBeDisabled();
  });

  it("should be enabled when not disabled", () => {
    render(<FormButton name="submit">Testando Botão</FormButton>);
    const button = screen.getByRole("button", { name: /testando botão/i });
    expect(button).toBeEnabled();
  });

  it("should call formHolder function on button click", () => {
    const fn = jest.fn();
    render(
      <FormFull onSubmit={fn}>
        <FormButton name="submit" actionType="submit">
          Testando Botão
        </FormButton>
      </FormFull>
    );

    const button = screen.getByRole("button", { name: /testando botão/i });
    userEvent.click(button);
    expect(fn).toHaveBeenCalledTimes(1);
  });

  it("should match snapshot", () => {
    render(<FormButton name="submit">Testando Botão</FormButton>);
    const button = screen.getByRole("button", { name: /testando botão/i });
    expect(button).toMatchSnapshot();
  });
});
