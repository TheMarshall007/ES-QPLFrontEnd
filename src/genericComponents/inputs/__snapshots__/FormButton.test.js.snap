// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genericComponents / inputs / <FormButton /> should match snapshot 1`] = `
<button
  class="btn btn-primary"
>
  Testando Bot√£o
</button>
`;
